hadolint: add page (#4782)
tldr-pages/tldr: pages/common/hadolint.md

# hadolint

> > Lints Dockerfiles to help you build [best practice](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices) Docker images.

Reviewer: Linter always try to send you to best practices. Maybe we can try this:
```suggestion
> Dockerfile linter.
> See also: <https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices>.
```
But, in a general way, we don't include more than 1 link in the pages. Another reviewer will comment about this.

> > More information: https://github.com/hadolint/hadolint.

Reviewer: ```suggestion
> More information: <https://github.com/hadolint/hadolint>.
```

> - Lint a Dockerfile, printing any output to STDOUT:

Reviewer: the `eslint` doesn't make any reference to `print to stdout`. I prefer to remove that
```suggestion
- Lint a Dockerfile:
```

Reviewer: Otherwise, we use this format for `stdout`:
```suggestion
- Lint a Dockerfile, printing the output to `stdout`:
```

Author: I don't think `stdout` is particularly important here

> `hadolint {{Dockerfile}}`

Reviewer: ```suggestion
`hadolint {{path/to/Dockerfile}}`
```

> - Lint a Dockerfile ignoring specific rules and outputting JSON:

Reviewer: I think you should split the format output from ignore rules. Which are the possible formats?

Author: The formats are `tty | json | checkstyle | codeclimate | codacy`
Not sure how these are typically displayed

Reviewer: Not sure, but you used less than 8 examples. Maybe is better add an extra example

Author: Added another example

> `hadolint --ignore {{DL3006}} --format json {{Dockerfile}}`

Reviewer: You need to say how to add a 2nd rule, could be a coma or space-separated list of rules, but for this command I saw this is the way:
```suggestion
`hadolint --ignore {{DL3006}} --ignore {{DL3008}} --format json {{path/to/Dockerfile}}`
```

Author: Tested and verified only way to pass through multiple rules is to use the `--include` flag multiple times

- Lint multiple Dockerfiles while specifying a trusted registry:

> `hadolint --trusted-registry {{my.company.com:500}} {{app/Dockerfile}} {{db/Dockerfile}}`

Reviewer: We use `example.com` for examples, the port is required?
```suggestion
`hadolint --trusted-registry {{example.com}}:{{500}} {{path/to/Dockerfile}} {{path/to/another/Dockerfile}}`
```

Author: `example.com` is good. The port isn't strictly required. The default docker registry port is `5000` (not `500` sorry).

Author: Actually, I think it's best if we show an example of each. Just updated it

# hadolint

> Dockerfile linter.
> More information: <https://github.com/hadolint/hadolint>.

- Lint a Dockerfile:

`hadolint {{path/to/Dockerfile}}`

- Lint a Dockerfile outputting a specific format:

`hadolint --format {{json}} {{path/to/Dockerfile}}`

- Lint a Dockerfile ignoring specific rules:

`hadolint --ignore {{DL3006}} --ignore {{DL3008}} {{path/to/Dockerfile}}`

> - Lint multiple Dockerfiles while specifying a trusted registries:

Reviewer: there are 2 registries. Maybe this is better:
```suggestion
- Lint multiple Dockerfiles using specific trusted registries:
```

Author: Agreed. Made the change

# hadolint

> Dockerfile linter.
> More information: <https://github.com/hadolint/hadolint>.

- Lint a Dockerfile:

`hadolint {{path/to/Dockerfile}}`

> - Lint a Dockerfile outputting in json:

Reviewer: ```suggestion
- Lint a Dockerfile, displaying the output in JSON format:
```

`hadolint --format {{json}} {{path/to/Dockerfile}}`

> - Lint a Dockerfile outputting in a codeclimate-friendly format:

Reviewer: ```suggestion
- Lint a Dockerfile, displaying the output in a specific format:
```

> `hadolint --format {{codeclimate}} {{path/to/Dockerfile}}`

Reviewer: I don't know if `codeclimate` is widely used, but I prefer a more generic example to display all the possible formats in the tldr pages:
```suggestion
`hadolint --format {{tty|json|checkstyle|codeclimate|codacy}} {{path/to/Dockerfile}}`
```

Author: Yeah. This is what I was looking for. This is a much better output. Thanks
